src/Repository/Repository.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace App\Repository;
+
+use App\Controllers\Home\HomeController;
+use App\Repository\DBConnexion;
+use PDO;
+
+abstract class Repository
+{
+    protected DBConnexion $dbConnection;
+    protected $table;
+    protected $entity;
+
+    public function __construct(DBConnexion $dbConnection, $table, $entity)
+    {
+        $this->dbConnection = $dbConnection;
+        $this->table = $table;
+        $this->entity = $entity;
+    }
+    //GET ALL
+    public function All(): array
+    {
+        try{
+            $req = $this->dbConnection->getPDO()->query("SELECT * FROM {$this->table}");
+            return $req->fetchAll(PDO::FETCH_CLASS, $this->entity);
+        }catch (\PDOException $exception){
+            return 'The following error has occured :' . $exception->getMessage() . ' at ligne ' . $exception->getLine() . ' in the following file: ' . $exception->getFile();
```

Comment:
This line does not respect your function return type. If you want return an error , you should throw an exception.